File: map-manager/src/scripts/export-openapi.ts
import { writeFileSync } from "fs";
import { resolve } from "path";
import { fileURLToPath } from "url";
import { createOpenApiDocument } from "../docs/openapi/document.js";

const __dirname = fileURLToPath(new URL(".", import.meta.url));
const document = createOpenApiDocument();
const outputPath = resolve(__dirname, "../../openapi.json");

writeFileSync(outputPath, JSON.stringify(document, null, 2), "utf8");
console.log(`OpenAPI schema successfully written to ${outputPath}`);

File: map-provider/src/scripts/export_openapi.py
import json
from pathlib import Path
from src.main import app

def export_openapi():
    schema = app.openapi()
    output_path = Path(__file__).resolve().parents[2] / "openapi.json"
    with open(output_path, "w", encoding="utf-8") as f:
        json.dump(schema, f, indent=2)
    print(f"OpenAPI schema successfully written to {output_path}")

if __name__ == "__main__":
    export_openapi()
